src/main/java/com/anthonymace/contacttasks/services/TasksService.java

```diff
@@ -0,0 +1,77 @@
+package com.anthonymace.contacttasks.services;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+
+@Service
+public class TasksService {
+
+    RestTemplate tasksRequest = new RestTemplate();
+    private String apiUrl = "https://api.infusionsoft.com/crm/rest/v1/%s";
+
+    @NotNull
+    private String accessToken;
+
+    public void setAccessToken(String accessToken) {
+        this.accessToken = accessToken;
+    }
+
+    public JSONArray getTasks(int contactId) {
+        String contactTasksUrl = String.format("tasks?contact_id=%d", contactId);
+        String tasksResourceUrl = String.format(apiUrl, contactTasksUrl);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON_UTF8));
+        headers.setBearerAuth(this.accessToken);
+
+        HttpEntity<String> request = new HttpEntity<>(headers);
+        ResponseEntity<String> tasksResponse =
+                tasksRequest.exchange(tasksResourceUrl, HttpMethod.GET, request, String.class);
+
+        JSONObject tasks = new JSONObject(tasksResponse.getBody());
```

Comment:
Could avoid using all these JSONObjects if we implemented serializable objects that new how to JSON-ify themselves.

Reply:
Ya I would spend the time making models for these. HttpEntity doesn't seem to play nicely with JSONObject anyway, will probably have to go to models or `Map<String, Object>` if taking this further.